\documentclass[11pt]{article}
\usepackage[utf8]{inputenc}
\usepackage{graphicx, verbatim}
\usepackage{hyperref}

\setlength{\textwidth}{6.5in} 
\setlength{\textheight}{9in}
\setlength{\oddsidemargin}{0in} 
\setlength{\evensidemargin}{0in}
\setlength{\topmargin}{-1.5cm}
%\VignetteIndexEntry{covEB}
\begin{document}

\begin{center}
{\bf \Large covEB package} \\
\end{center}
\SweaveOpts{keep.source=TRUE}
\DefineVerbatimEnvironment{Sinput}{Verbatim}{xleftmargin=2em}
\DefineVerbatimEnvironment{SOutput}{Verbatim}{xleftmargin=2em}
\DefineVerbatimEnvironment{Scode}{Verbatim}{xleftmargin=2em}
\fvset{listparameters={\setlength{\topsep}{0pt}}}
\renewenvironment{Schunk}{\vspace{\topsep}}{\vspace{\topsep}}






\section{Introduction}
<<echo=FALSE>>=
options(width=60)
@
Correlation matrices are important in inferring relationships and networks between regulatory or signalling elements. By necessity sample sizes for experiments are small, meaning that these correlations can be difficult to estimate. At a genome-wide scale estimation of correlation matrices can also be computationally demanding.
This package provides an empirical Bayes approach to improve covariance estimates for gene expression, where we assume the covariance matrix takes a block diagonal form. 

\section{A Simple Example}
We show a simple example of how to run the empirical Bayes estimation, these are trivial examples but serve to illustrate the syntax and parameters of the function. We use the package mvtnorm to simulate data from a multivariate normal distribution.
<<>>=
 library(mvtnorm)
 library(covEB)
 library(igraph)
 	sigma <- matrix(c(4,2,2,3), ncol=2)
	x <- rmvnorm(n=500, mean=c(1,2), sigma=sigma)
 
 	samplecov<-cov(x)
 
 	test<-covEB(samplecov,delta=0.05,shift=0.025,startlambda=0.4)


@
In this example there are three parameters excluding the covariance matrix, samplecov. These are delta, shift and startlambda. These three parameters together control the correlation ranges that define each of the blocks for which the empirical Bayes estimate is calculated. covEB is a novel iterative block diagonal algorithm for estimating the correlation matrix. Instead of one prior matrix for the full correlation matrix, we identify blocks of correlated elements for a given correlation range. Consequently we use a range of shrinkage values ($\theta's$) for each block and calculate the empirical Bayes estimate for each of these. We use a range to allow for noise in the data for example, if we assume the correlation for a block $b$ is $\theta_{b}$ and we there is noise in our data of $delta$ then the range of correlation values would be $(\theta_{b}-delta/2,\theta_{b} +delta/2)$. Values outside the current range are set to zero to determine the diagonal block structure.
From a starting threshold (startlambda) the two user selected parameters determine the size of the interval (delta) and the size of the shift in moving the starting threshold (shift). This shift value means that the range values used are overlapping rather than mutually exclusive. The block diagonal structure of the correlation matrix is determined using the current range and the empirical Bayesian estimation for each block performed separately. The final estimate is an average of each non zero estimation. Setting $startlambda>0$ will aid computation time as any correlations under this level are set to zero and may also be used to reduce noise in the data, by assuming any correlation less than startlambda is noise.

\section{References}
Champion, C. J. (2003). Empirical Bayesian estimation of normal variances and covariances. Journal of Multivariate Analysis, 87(1), 60Ð79. http://doi.org/10.1016/S0047-259X(02)00076-3




\end{document}